### ISSUE TITLE
NullPointerException thrown when renaming class together with symbol's file

### ISSUE THREAD

**Issue by miqm** [dotnet/roslyn#46192]
**Version Used**: 16.7-preview6, 16.7-preview5, 16.7-preview4

**Steps to Reproduce**:

1. Rename a class from other file than symbol's with Rename Symbol's file option checked

**Expected Behavior**:
Class and file should be renamed.
**Actual Behavior**:
NullPointerException is thrown.

Stack trace extracted from activity log:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProjectItem.<>c__DisplayClass47_0.<Save>b__0()
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProject.<>c__DisplayClass73_0.<<Mutate>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.<>c__DisplayClass565_0.<<Mutate>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.ProjectSystem.ProjectMultiThreadedService.ExecuteSynchronously(Func`1 asyncAction)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.Mutate(Func`1 asyncAction, Boolean includesFileSystemChange, String telemetryActivity)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProject.Mutate(Func`1 asyncAction, Boolean includesFileSystemChange, String telemetryActivity)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProjectItem.Save(String fileName)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.ApplyDocumentInfoChanged(DocumentId documentId, DocumentInfo updatedInfo)
   at Microsoft.CodeAnalysis.Workspace.ApplyChangedDocument(ProjectChanges projectChanges, DocumentId documentId)
   at Microsoft.CodeAnalysis.Workspace.ApplyProjectChanges(ProjectChanges projectChanges)
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.ApplyRename(Solution newSolution, IWaitContext waitContext)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.CommitCore(IWaitContext waitContext, Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.<>c__DisplayClass78_0.<CommitWorker>b__0(IWaitContext waitContext)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Utilities.VisualStudioWaitIndicator.Wait(String title, String message, Boolean allowCancel, Boolean showProgress, Action`1 action)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.CommitWorker(Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext context)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.<>c__DisplayClass14_1`1.<ExecuteStartingAtIndex>b__0()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call, Predicate`1 exceptionFilter)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject
```

This might relate to recent changes in this functionality: #44576

**Comment by miqm** [comment]
Additional observation - rename works when symbol's file is opened in editor.

**Comment by miqm** [comment]
@jinujoseph Is the fix is really going live with 16.8 and the buggy feature with 16.7?

**Comment by jinujoseph** [comment]
@miqm , 16.7 is only taking critical bug fixes at this point. once @ryzngard investigates the issue and find the fix, we can backport the fix to 16.7

**Comment by miqm** [comment]
@jinujoseph In my humble opinion, VS throwing NPE during a class rename (which I think is a common thing developers do) is a critical bug, especially that it was working before (not entirely correct as intended, but NPE was not thrown)

**Comment by ryzngard** [comment]
@miqm I was unable to reproduce this with a basic example using 16.7 Preview 5. Do you have more information or a solution I can use?

**Comment by miqm** [comment]
@ryzngard 
![image](https://user-images.githubusercontent.com/7921224/88718146-d079f180-d121-11ea-82ee-2d604c414ab3.png)
Pressed Enter
![image](https://user-images.githubusercontent.com/7921224/88718355-25b60300-d122-11ea-9c73-067c362e9db9.png)

I also now noticed, when I click "Apply" on the dialog on the right, VS freezes and crashes.

This is on 16.7-preview6

[ActivityLog.xml.txt](https://github.com/dotnet/roslyn/files/4991043/ActivityLog.xml.txt) (sorry for the TXT, github does not allow xml files as attachment)

**Comment by ryzngard** [comment]
Hmm... still not seeing the repro locally. That said, something odd about what's going on that I noticed is different. Your file is already called `Context.cs`, which means you should be seeing this: 


![image](https://user-images.githubusercontent.com/475144/88719588-612bdf80-d0d8-11ea-9475-5d623ad0bb87.png)

**Comment by miqm** [comment]
Why? I'm renaming from Context to Context2, so when renaming, I shouldn't see that dialog.

**Comment by ryzngard** [comment]
Oh nm, I see you had `Context` and renamed to `Context2`. Still works correctly for me. I misinterpreted that you were showing after you typed the name change not before.

**Comment by miqm** [comment]
But did you rename it from Context.cs file or from Program.cs file? Also, make sure you **do not** have Context.cs file opened in editor.

**Comment by ryzngard** [comment]
> Also, make sure you do not have Context.cs file opened in editor

There's the difference :) Have a repro now

**Comment by ryzngard** [comment]
Thanks for the help @miqm , PR with a workaround/fix is up. We'll need to investigate a few things to determine if this will go into 16.7 or 16.8.

**Comment by miqm** [comment]
I wonder how fast people will start reporting this issue from production release...

**Comment by ryzngard** [comment]
Fixed in 16.8 P2
